### DISCUSSION TITLE
multi-gpu modify batchsize

### DISCUSSION THREAD

**Discussion by shangyihao** [Lightning-AI/pytorch-lightning#7450] [DDP / multi-GPU / multi-node]
when i use Multi-GPU training in pytorch lightning, like Trainer(gpus=[1,2], accelerator='dp') with
![image](https://user-images.githubusercontent.com/71238607/117565585-9001c800-b0e4-11eb-927b-7e73ed4ed2d7.png)
what or where can i do to modify  the effective batch size for distributed training?
thx

**Comment by Borda** [comment]
you have set `batch_size` already, if you want to adjust the size according to actual GPU, you can use [auto_scale_batch_size](https://pytorch-lightning.readthedocs.io/en/latest/common/trainer.html#auto-scale-batch-size)
Moreover, I see that your `train_dataloder` returns a sampler instead of dataloader instance (or list f instances)
